src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSparkUnitTest.java

```diff
@@ -0,0 +1,287 @@
+
+        editDistancesBetweenSeq = new ArrayList<>();
+        final File correctedFile = new File(workingDir, correctedFileName);
+        final File actualFilterPassingFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGAFilter(sgaPath, correctedFile, workingDir, threads).fileNameToReturn);
+        final File expectedFilterPassingFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.filter.pass.fa");
+        final String filterPassingFileName = compareNamesAndComputeSeqEditDist(actualFilterPassingFile, expectedFilterPassingFile, true, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        editDistancesBetweenSeq = new ArrayList<>();
+        final File filterPassingFile = new File(workingDir, filterPassingFileName);
+        final File actualRmdupFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGARmDuplicate(sgaPath, filterPassingFile, workingDir, threads, indexer, indexerArgs).fileNameToReturn);
+        final File expectedRmdupFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.filter.pass.rmdup.fa");
+        final String duplicateRemovedFileName = compareNamesAndComputeSeqEditDist(actualRmdupFile, expectedRmdupFile, false, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        editDistancesBetweenSeq = new ArrayList<>();
+        final File duplicateRemovedFile = new File(workingDir, duplicateRemovedFileName);
+        final File actualMergedFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGAFMMerge(sgaPath, duplicateRemovedFile, workingDir, threads, indexer, indexerArgs).fileNameToReturn);
+        final File expectedMergedFile = new File(TEST_DATA_DIR, filenamePrefix + ".pp.ec.filter.pass.rmdup.merged.fa");
+        final String mergedFileName = compareNamesAndComputeSeqEditDist(actualMergedFile, expectedMergedFile, false, editDistancesBetweenSeq);
+        for(final Integer d : editDistancesBetweenSeq){ Assert.assertEquals(d, zero); }
+
+        // final assembled contig test
+        final File mergedFile = new File(workingDir, mergedFileName);
+        final File actualAssembledContigsFile = new File(workingDir, RunSGAViaProcessBuilderOnSpark.runSGAOverlapAndAssemble(sgaPath, mergedFile, workingDir, threads).fileNameToReturn);
+        final RunSGAViaProcessBuilderOnSpark.ContigsCollection actualAssembledContigs = new RunSGAViaProcessBuilderOnSpark.ContigsCollection(actualAssembledContigsFile);
+        final RunSGAViaProcessBuilderOnSpark.ContigsCollection expectedAssembledContigs = new RunSGAViaProcessBuilderOnSpark.ContigsCollection(expectedAssembledContigFile);
+        Assert.assertTrue( compareContigs(actualAssembledContigs, expectedAssembledContigs) );
+    }
+
+    private static boolean compareContigs(final RunSGAViaProcessBuilderOnSpark.ContigsCollection actualAssembledContigs,
```

Comment:
Rather than have this as a boolean returning method, I think this would be better as a void-returning method that does an `Assert` on each thing it is comparing. That way if a test fails you can get the exact contig that failed from the assertion failure, rather than having to debug your test just to get going.